packages/did/src/Did.utils.ts

```diff
@@ -43,30 +46,54 @@ type IDidParsingResult = {
   version: number
   type: 'light' | 'full'
   address: KiltAddress
+  queryParameters?: Record<string, string>
   fragment?: UriFragment
   authKeyTypeEncoding?: string
   encodedDetails?: string
 }
 
+// Exports the params section of a DID URL as a map.
+// If multiple keys are present, only the first one is returned.
+function exportQueryParamsFromUri(didUri: DidUrl): Record<string, string> {
+  const urlified = new URL(didUri)
+  const params: Record<string, string> = {}
+  urlified.searchParams.forEach((value, key) => {
```

Comment:
The `URLSearchParams` only exposes a `forEach` function. Using `reduce` would imply we first have to convert that to an array, and then call `reduce` on it. I though it would be overkill for such a small thing. What do you think? Also, using `reduce` does not spare us from having to add the key to the set only if not present, otherwise we would take the last value instead of the first one.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/799/commits/1cfe243e301e47b7601be5abd4a8ae6314aea1b1